modules/nux/modules.php

```diff
@@ -245,6 +245,293 @@ public function process() {
+                           $res = Hm_SMTP_List::del($this->smtp_server_id);
+                           if ($res) {
+                               Hm_SMTP_List::forget_credentials($this->smtp_server_id);
+                               $smtp_servers = Hm_SMTP_List::dump(false, true);
+                               $this->user_config->set('smtp_servers', $smtp_servers);
+                           }
+                       }
+                        return;
+                     }
+                 }
+    
+                 if($form['nux_config_is_sender'] && $form['nux_config_is_receiver'] && $form['nux_create_profile'] && isset($this->imap_server_id) && isset($this->smtp_server_id)) {
+                     $this->saveProfile(
+                        $form['nux_config_profile_name'],
+                        $form['nux_profile_signature'],
+                        $form['nux_profile_reply_to'],
+                        $form['nux_profile_is_default'],
+                        $form['nux_config_email'],
+                        $form['nux_config_imap_address']
+                        );
+                 }
+    
+                 Hm_Msgs::add("Server saved");
+                 $this->out('just_saved_credentials', $this->just_saved_credentials);
+            }
+        }
+    }
+
+    public function saveProfile($name, $signature, $replyTo, $isDefault, $email, $serverMail){
+        $profile = array(
+                         'name' => $name,
+                         'sig' => $signature,
+                         'smtp_id' => $this->imap_server_id,
+                         'replyto' => $replyTo,
+                         'default' => $isDefault,
+                         'address' => $email,
+                         'server' =>  $serverMail,
+                         'user' => $email,
+                         'type' => 'imap'
+                     );
+
+          $profiles = new Hm_Profiles($this);
+          $profiles->add($profile);
+          $this->session->record_unsaved('Profile added');
+
+          $profiles->save($this->user_config);
+          $user_data = $this->user_config->dump();
+          $this->session->set('user_data', $user_data);
+    }
+
+    public function connectToIMAP($address, $name, $port, $user, $pass, $tls, $imap_sieve_host, $enableSieve = false, $errno = null, $errstr = null) {
+        if ($con = fsockopen($address, $port, $errno, $errstr, 5)) {
+              $imap_list = array(
+                  'name' => $name,
+                  'server' => $address,
+                  'hide' => false,
+                  'port' => $port,
+                  'user' => $user,
+                  'pass' => $pass,
+                  'tls' => $tls);
+
+              if ($this->module_is_supported('sievefilters') && $this->user_config->get('enable_sieve_filter_setting', true) && $enableSieve) {
+                  $imap_list['sieve_config_host'] = $imap_sieve_host;
+
+                   require_once VENDOR_PATH . 'autoload.php';
```

Comment:
We shouldn't randomly autoload this. It should already be autoloaded in the lib/framework file. If not, check latest master.